### PR TITLE
[docs] improve bart/marian/mBART/pegasus docs

### LINKED ISSUE #8416: Does MBartTokenizer remove the parameter decoder_input_ids?
## Environment info

     
- `transformers` version:3.4.0
- Platform:Google Colab
- Python version:3.7
- PyTorch version (GPU?):1.7.0+cu101
- Tensorflow version (GPU?):2.x
- Using GPU in script?: no
- Using distributed or parallel set-up in script?:no

## Information

Model I am using (Bert, XLNet ...): mbart

The problem arises when using:
* [x] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce
```python
example_english_phrase = "UN Chief Says There Is No Military Solution in Syria"
expected_translation_romanian = "Şeful ONU declară că nu există o soluţie militară în Siria"
batch = MBartTokenizer.from_pretrained('facebook/mbart-large-cc25').prepare_seq2seq_batch(example_english_phrase, src_lang="en_XX", tgt_lang="ro_RO", tgt_texts=expected_translation_romanian)
input_ids = batch["input_ids"]
target_ids = batch["decoder_input_ids"]
```
```python
KeyError                                  Traceback (most recent call last)
<ipython-input-11-b3eedaf10c3e> in <module>()
      3 batch = MBartTokenizer.from_pretrained('facebook/mbart-large-en-ro').prepare_seq2seq_batch(example_english_phrase, src_lang="en_XX", tgt_lang="ro_RO", tgt_texts=expected_translation_romanian)
      4 input_ids = batch["input_ids"]
----> 5 target_ids = batch["decoder_input_ids"]

/usr/local/lib/python3.6/dist-packages/transformers/tokenization_utils_base.py in __getitem__(self, item)
    232         """
    233         if isinstance(item, str):
--> 234             return self.data[item]
    235         elif self._encodings is not None:
    236             return self._encodings[item]

KeyError: 'decoder_input_ids'
```

### PR DESCRIPTION
**Pull request by sshleifer**
+ Give example of bart mask filling
+ Link to training scripts where applicable
+ Clarify Marian naming scheme a bit.

### CODE REVIEW

File: docs/source/model_doc/marian.rst [hunk 1]
```diff
@@ -99,16 +130,23 @@ For example:
...
-    suffix = [x.split('/')[1] for x in model_ids]
-    multi_models = [f'{org}/{s}' for s in suffix if s != s.lower()]
+
+Example of translating english to many romance languages, using old-style 2 character language codes
+
+
+.. code-block::python
+
+    from transformers import MarianMTModel, MarianTokenizer
+     src_text = [ '>>fr<< this is a sentence in english that we want to translate to french', '>>pt<< This should go to portuguese', '>>es<< And this to Spanish']
```
Comment (sgugger) [inline docs/source/model_doc/marian.rst:142]: You got some bad indent here. It then made the rest of the code badly formatted, so you need to re-add the new lines at the right places.

**Review by sgugger — APPROVED** [review]
Thanks for cleaning the docs!

**Review by LysandreJik — APPROVED** [review]
LGTM, thanks @sshleifer!

File: docs/source/model_doc/marian.rst [hunk 2]
```diff
@@ -143,9 +143,8 @@ Example of translating english to many romance languages, using old-style 2 char
 
     model_name = 'Helsinki-NLP/opus-mt-en-ROMANCE' tokenizer = MarianTokenizer.from_pretrained(model_name)
     print(tokenizer.supported_language_codes) model = MarianMTModel.from_pretrained(model_name) translated =
-    model.generate(**tokenizer.prepare_seq2seq_batch(src_text)) tgt_text = [tokenizer.decode(t,
-    skip_special_tokens=True) for t in translated] # ["c'est une phrase en anglais que nous voulons traduire en
-    français", # 'Isto deve ir para o português.', # 'Y esto al español']
+    model.generate(**tokenizer.prepare_seq2seq_batch(src_text)) tgt_text = [tokenizer.decode(t, skip_special_tokens=True) for t in translated]
```
Comment (sgugger) [inline docs/source/model_doc/marian.rst:146]: Mmmm, this did not fix anything...

Comment (sshleifer) [reply docs/source/model_doc/marian.rst:146]: darn
will get it now. my bad

**Pull request merged**